ota-plus-web/app/com/advancedtelematic/LoggingFilter.scala

```diff
@@ -4,44 +4,61 @@ import javax.inject.Inject
 
 import com.advancedtelematic.api.RemoteApiError
 import play.api.Logger
+import play.api.http.Status
 import play.api.mvc.{Filter, RequestHeader, Result}
 import _root_.akka.stream.Materializer
 
 import scala.concurrent.{ExecutionContext, Future}
 import scala.util.{Failure, Success}
 
 class LoggingFilter @Inject()(implicit val mat: Materializer, ec: ExecutionContext) extends Filter {
-  lazy val serviceName = sys.env.getOrElse("SERVICE_NAME", "ota-plus-web")
+
+  import LoggingFilter._
 
   def apply(nextFilter: RequestHeader => Future[Result])
            (requestHeader: RequestHeader): Future[Result] = {
+    val message = getMessage(requestHeader, _: Int, _: Long, System.currentTimeMillis)
 
-    val startTime = System.currentTimeMillis
     nextFilter(requestHeader) andThen {
       case Success(result) =>
-        logRequest(requestHeader, result.header.status, startTime)
+        logger.info(message(result.header.status, result.body.contentLength.getOrElse(0)))
       case Failure(RemoteApiError(result, _)) =>
-        logRequest(requestHeader, result.header.status, startTime)
+        logger.info(message(result.header.status, result.body.contentLength.getOrElse(0)))
       case _ =>
-        logRequest(requestHeader, 500, startTime)
+        logger.info(message(Status.INTERNAL_SERVER_ERROR, 0))
     }
   }
+}
+
+object LoggingFilter {
+  
+  private val logger = Logger("json-accesslog")
 
-  protected def logRequest(requestHeader: RequestHeader, resultCode: Int, startTime: Long): Unit = {
+  val serviceName = sys.env.getOrElse("SERVICE_NAME", "ota-plus-web")
+
+  def getMessage(requestHeader: RequestHeader,
+                 responseCode: Int,
+                 responseContentLength: Long,
+                 startTime: Long): String = {
     val endTime = System.currentTimeMillis
     val serviceTime = endTime - startTime
 
-    val metrics = Map(
-      "service_name" -> serviceName,
+    val data = List(
```

Comment:
please check https://github.com/advancedtelematic/libats/blob/master/libats-http/src/main/scala/com/advancedtelematic/libats/http/LogDirectives.scala. Logging filter should have similar behaviour.

Reply:
Done.